src/v4.js

```diff
@@ -19,12 +19,14 @@ function v4(options, buf, offset) {
 
   // Copy bytes to buffer, if provided
   if (buf) {
-    for (let ii = 0; ii < 16; ++ii) {
-      buf[i + ii] = rnds[ii];
+    for (let i = 0; i < 16; ++i) {
+      buf[start + i] = rnds[i];
     }
+
+    return buf;
   }
 
-  return buf || bytesToUuid(rnds);
+  return bytesToUuid(rnds);
```

Comment:
Gotta return `buf` if it's passed in.

```suggestion
  return buf || bytesToUuid(rnds);
```

Reply:
@broofa sure, I can add test for this case.

But the code can be seen that the buffer is returned after it is filled:
https://github.com/uuidjs/uuid/pull/435/files#diff-8988f35cbc1e6b5ea7f9cb52a4bfa0f2R26

Reply:
@broofa there is already tests for this case:

https://github.com/uuidjs/uuid/blob/master/test/unit/v4.test.js#L46-L73

Reply:
@awwit I just realized that we currently do two things: fill the passed buffer and return it. But we only test for filling, not for the returned buffer.

Since we don't know if anybody relies on the fact, that our methods fill the provided buffer and return it, it would be a good opportunity to add another test case that test for the return value when a buffer is passed.